CleverPush/Source/CleverPushInstance.m

```diff
@@ -402,6 +414,9 @@ - (void)initWithChannelId {
                 if (handleSubscribedInternal) {
                     handleSubscribedInternal(subscriptionId);
                 }
+                if (handleInitialized) {
```

Comment:
This needs to be moved, right now we only call handleInitialized when subscriptionId is non-nil which is not what we want.
We want to inform the customer if the init has been successful or not. So also on error we need to call handleInitialized(NO) and provide customer with the error there. For example if `getChannelConfig fails in the initialisation process.

Reply:
In `getChannelConfing`, we have two main functions.

1. `[self getChannelConfigFromChannelId:configPath];`
2. `[self getChannelConfigFromBundleId:configPath];`

So I think that we need to handle success and failure as blocks in both functions.

That's why I keep the callback success or error message with the error message as well.
If it is successful, the error message will be null.
If it is false, the error message will be the same as what we are displaying in the log.

Reply:
tested with the below scenarios (if I set the init callback function while using the init method).

1. First, I don't set the channelID, and we are getting false in the callback of the init listener.
2. Second, I set the channel ID, and we are getting success in the callback of the init listener.